add audit log (#1229)
vesoft-inc/nebula-docs: docs-2.0/5.configurations-and-logs/2.log-management/audit-log.md

# Audit logs

> The audit logs store all operations received by graph service in categories, then provide the logs for users to track specific types of operations as needed.

Reviewer: The audit logs --？Nebula Graph audit logs?

|`dcl`|`CREATE USER`、`GRANT ROLE`、`REVOKE ROLE`、`CHANGE PASSWORD`、`ALTER USER`、`DROP USER`、`CREATE SNAPSHOT`、`DROP SNAPSHOT`、`ADD LISTENER`、`REMOVE LISTENER`、`BALANCE`、`SUBMIT JOB`、`STOP JOB`、`RECOVER JOB`、`ADD DRAINER`、`REMOVE DRAINER`|Logs the information about DCL statements.|
|`util`|`SHOW HOSTS`、`SHOW USERS`、`SHOW ROLES`、`SHOW SNAPSHOTS`、`SHOW SPACES`、`SHOW PARTS`、`SHOW TAGS`、`SHOW EDGES`、`SHOW INDEXES`、`SHOW CREATE SPACE`、`SHOW CREATE TAG/EDGE`、`SHOW CREATE INDEX`、`SHOW INDEX STATUS`、`SHOW LISTENER`、`SHOW TEXT SEARCH CLIENTS`、`SHOW DRAINER CLIENTS`、`SHOW FULLTEXT INDEXES`、`SHOW CONFIGS`、`SHOW CHARSET`、`SHOW COLLATION`、`SHOW STATS`、`SHOW SESSIONS`、`SHOW META LEADER`、`SHOW DRAINERS`、`SHOW QUERIES`、`SHOW JOB`、`SHOW JOBS`、`DESCRIBE INDEX`、`DESCRIBE EDGE`、`DESCRIBE TAG`、`DESCRIBE SPACE`、`DESCRIBE USER`、`USE SPACE`、`SIGN IN TEXT SERVICE`、`SIGN OUT TEXT SERVICE`、`SIGN IN DRAINER SERVICE`、`SIGN OUT DRAINER SERVICE`、`EXPLAIN`、`PROFILE`、`KILL QUERY`、`DOWNLOAD HDFS`、`INGEST`|Logs the information about util statements. |
|`unknown`|-|Logs the information about unrecognized statements.|

## Configure audit logs

> You need to configure the configuration of the graph service to view audit logs. The default file path of configuration is `/usr/local/nebula/etc/nebula-graphd.conf`.

Reviewer: ```suggestion
You need to configure the graph service file to view audit logs. The default file path of configuration is `/usr/local/nebula/etc/nebula-graphd.conf`.
```

!!! note

    After modifying the configuration, you need to restart the graph service to take effect.

Parameter descriptions are as follows:

|Parameter|Predefined value|Description|
|:--|:--|:--|
| `enable_audit` | `false` | Whether or not to enable audit logs. |

> | `audit_log_handler` | `file` | Where the audit logs will be written. Optional values are `file`（local file） and `es`(Elasticsearch). |

Reviewer: ```suggestion
| `audit_log_handler` | `file` | Where will the audit logs be written. Optional values are `file`（local file） and `es`(Elasticsearch). |
```

> | `audit_log_file` | `./logs/audit/audit.log` | Takes effect only when `audit_log_handler=file`. The path for storing audit logs. The value can be absolute or relative. |

Reviewer: ```suggestion
| `audit_log_file` | `./logs/audit/audit.log` | Takes effect only when `audit_log_handler=file` is set. The path for storing audit logs, the value of which can be absolute or relative. |
```

> | `audit_log_strategy` | `synchronous` | Takes effect only when `audit_log_handler=file`. The strategy of the audit logs. Optional values are `asynchronous` and `synchronous`. When set to `asynchronous`, the log using memory buffer does not block the main thread, but the log may be missing due to insufficient cache. When set to `synchronous`, the log directly to file, flush and sync every event. |

Reviewer: ```suggestion
| `audit_log_strategy` | `synchronous` | Sets the method to synchronize audit logs. Takes effect only when `audit_log_handler=file` is set.  Optional values are `asynchronous` and `synchronous`. When `asynchronous` is set, log events are cached in memory and do not block the main thread, but may result in missing logs due to insufficient cache. When `synchronous` is set, log events are refreshed and synchronized to the file each time. |
```

> | `audit_log_max_buffer_size` | `1048576` |Take effect only when `audit_log_handler=file` and `audit_log_strategy=asynchronous`. The size of the memory buffer used for logging. Unit: bytes.  |

Reviewer: ditto 
pls check below content

```bash
<AUDIT_RECORD
  CATEGORY="util"
  TIMESTAMP="2022-04-07 02:31:38"
  TERMINAL=""
  CONNECTION_ID="1649298693144580"
  CONNECTION_STATUS="0"
  CONNECTION_MESSAGE=""
  USER="root"
  CLIENT_HOST="127.0.0.1"
  HOST="192.168.8.111"
  SPACE=""
  QUERY="use basketballplayer1"
  QUERY_STATUS="-1005"
  QUERY_MESSAGE="SpaceNotFound: "
/>
<AUDIT_RECORD
  CATEGORY="util"
  TIMESTAMP="2022-04-07 02:31:39"
  TERMINAL=""
  CONNECTION_ID="1649298693144580"
  CONNECTION_STATUS="0"
  CONNECTION_MESSAGE=""
  USER="root"
  CLIENT_HOST="127.0.0.1"
  HOST="192.168.8.111"
  SPACE=""
  QUERY="use basketballplayer"
  QUERY_STATUS="0"
  QUERY_MESSAGE=""
/>
```

|Field|Description|
|:--|:--|
|`CATEGORY`| The category of the audit logs.|
|`TIMESTAMP`| The generation time of the audit logs. |
|`TERMINAL`| The reserved field.|
|`CONNECTION_ID`| The session ID of the connection. |
|`CONNECTION_STATUS`| The status of the connection. `0` indicates success, and other numbers indicate different error messages.|
|`CONNECTION_MESSAGE`| An error message is displayed when the connection fails.|

> |`USER`| The user name of the connection. |

Reviewer: The user name for the Nebula Graph connection?

|`CLIENT_HOST`| The IP address of the client.|
|`HOST`| The IP address of the host. |

> |`SPACE`| The graph space to perform the query.|

Reviewer: ```suggestion
|`SPACE`| The graph space where you perform queries.|
```
